src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1111,16 +1109,37 @@ class SQLTargetWHandle : public StorageTargetWHandle {
         sha512Hash = hash.HashString();
       }
     }
-    auto statement = db_.prepareStatement<std::string, SQLZeroBlob>(
-        "INSERT OR REPLACE INTO target_images (filename, image_data, sha256, sha512) VALUES (?, ?, ?, ?);",
-        target_.filename(), SQLZeroBlob{target_.length()}, sha256Hash, sha512Hash);
+    auto statement = db_.prepareStatement<std::string>("SELECT rowid FROM target_images_data WHERE filename = ?;",
+                                                       target_.filename());
+
+    if (statement.step() == SQLITE_ROW) {
+      statement =
+          db_.prepareStatement<std::string>("DELETE FROM target_images_data WHERE filename=?;", target_.filename());
```

Comment:
Could you also explain, why we need an explicit delete here? I thought REPLACE will do the same.

Reply:
Yes you are right it is odd.